packages/@lwc/engine-core/src/framework/hot-swaps.ts

```diff
@@ -148,24 +148,26 @@ export function setActiveVM(vm: VM) {
     activeComponents.add(Ctor, vm);
 
     // tracking active template
-    const tpl = vm.cmpTemplate;
-    if (tpl) {
+    const template = vm.cmpTemplate;
+    if (!isNull(template)) {
         // this will allow us to keep track of the templates that are
         // being used by a hot component
-        activeTemplates.add(tpl, vm);
-
-        // tracking active styles associated to template
-        const stylesheets = tpl.stylesheets;
-        if (!isUndefined(stylesheets)) {
-            for (const stylesheet of flattenStylesheets(stylesheets)) {
-                // this is necessary because we don't hold the list of styles
-                // in the vm, we only hold the selected (already swapped template)
-                // but the styles attached to the template might not be the actual
-                // active ones, but the swapped versions of those.
-                const swappedStylesheet = getStyleOrSwappedStyle(stylesheet);
-                // this will allow us to keep track of the stylesheet that are
-                // being used by a hot component
-                activeStyles.add(swappedStylesheet, vm);
+        activeTemplates.add(template, vm);
+
+        // Tracking active styles from the template or the VM. `template.stylesheets` are implicitly associated
+        // (e.g. `foo.css` associated with `foo.html`), whereas `vm.stylesheets` are from `static stylesheets`.
+        for (const stylesheets of [template.stylesheets, vm.stylesheets]) {
```

Comment:
How does this perf compare to creating a helper function and doing `flatten(template.stylesheets); flatten(vm.stylesheets)`? (Or even just making it WET.)

Reply:
Creating a new array on-demand is definitely not ideal perf-wise. I kind of didn't care because this is dev-only code.

However, we may as well optimize this; it may become hot in a dev environment with lots of components. I'll do that.